### PR TITLE
Creating a simple grep example

### PR DESCRIPTION
**Pull request by mattnenterprise**
@jdoliner I am trying to create a simple grep example. It currently seems to be working, but I would like some input on improving it. I added the correct files for running the job and documentation in the QUICKSTART.md. I have a few questions about the pipeline system.

1. What are the definitions of a pipeline and a job ? How to they relate to each other ?

2. How do I make this truly distributed if I have a lot of files, and want to process them in multiple containers ?

3. It currently looks like the job-shim binary is required to be in the container. Will this always be required ?

### CODE REVIEW

**Comment by mattnenterprise** [comment]
This relates to #157

**Comment by jdoliner** [comment]
Nice! Thanks so much for this @mattnenterprise will merge.

> What are the definitions of a pipeline and a job ? How to they relate to each other ?

If you're familiar with k8s, pipelines are to jobs as replicationcontrollers are to pods.
Jobs are simpler, and are the basic unit of computation in Pachyderm.
They take a set of commits as inputs and create a single output commit.
The output commit is guaranteed to exist and be finished once the job has completed.

Pipelines are a way to automate jobs. They take a set of repos as inputs. Any time a commit is made to one of those repos Pachyderm will create a job which processes the new commit and ultimately creates a new commit in the pipelines output repo. 

> How do I make this truly distributed if I have a lot of files, and want to process them in multiple containers ?
It's already truly distributed. Your code doesn't need to do anything special to be distributed. Jobs (and Pipelines) have a `Shard` parameter which defaults to `1`. If you crank this up then Pachyderm will create multiple copies of the container and each of them will see a different subset of the files.


> It currently looks like the job-shim binary is required to be in the container. Will this always be required ?
No, this is incredibly annoying and needs to go away sooner rather than later. Long term I want to create pods with 2 containers, one which mounts the fuse volume and once which is the users container and runs code. Right now there's no way to propagate a fuse mount between containers but that should be changing soon (it might have already).
So tl;dr we're waiting on mount propagation in Docker before we can get rid of this.
